packages/zoltan2/src/algorithms/color/Zoltan2_AlgHybridGMB.hpp

```diff
@@ -0,0 +1,458 @@
+#ifndef _ZOLTAN2_ALGHYBRIDGMB_HPP_
+#define _ZOLTAN2_ALGHYBRIDGMB_HPP_
+
+#include <vector>
+#include <unordered_map>
+#include <iostream>
+#include <queue>
+#include <unistd.h>
+
+#include <Zoltan2_Algorithm.hpp>
+#include <Zoltan2_GraphModel.hpp>
+#include <Zoltan2_ColoringSolution.hpp>
+#include <Zoltan2_Util.hpp>
+#include <Zoltan2_TPLTraits.hpp>
+
+#include "Tpetra_Core.hpp"
+#include "Teuchos_RCP.hpp"
+#include "Tpetra_Import.hpp"
+#include "Tpetra_FEMultiVector.hpp"
+
+#include <KokkosKernels_Handle.hpp>
+#include "KokkosKernels_IOUtils.hpp"
+#include "KokkosGraph_Distance1Color.hpp"
+#include "KokkosGraph_Distance1ColorHandle.hpp"
+
+#include "zz_rand.h"
+//////////////////////////////////////////////
+//! \file Zoltan2_AlgHybridGMB.hpp
+//! \brief A hybrid version of the framework proposed by Gebremedhin, Manne, 
+//!        and Boman
+
+namespace Zoltan2 {
+
+template <typename Adapter>
+class AlgHybridGMB : public Algorithm<Adapter>
+{
+  public:
+  
+    typedef typename Adapter::lno_t lno_t;
+    typedef typename Adapter::gno_t gno_t;
+    typedef typename Adapter::offset_t offset_t;
+    typedef typename Adapter::scalar_t scalar_t;
+    typedef typename Adapter::base_adapter_t base_adapter_t;
+    typedef Tpetra::Map<lno_t, gno_t> map_t;
+    typedef int femv_scalar_t;
+    typedef Tpetra::FEMultiVector<femv_scalar_t, lno_t, gno_t> femv_t;
+    
+  private:
+
+    void buildModel(modelFlag_t &flags); 
+
+    //function to invoke KokkosKernels distance-1 coloring    
+    template <class ExecutionSpace, typename TempMemorySpace, 
+              typename MemorySpace>
+    void colorInterior(const size_t nVtx, 
+                       Kokkos::View<lno_t*, Kokkos::Device<ExecutionSpace,MemorySpace> > adjs_view,
+                       Kokkos::View<offset_t*, Kokkos::Device<ExecutionSpace,MemorySpace> > offset_view, 
+                       Teuchos::ArrayRCP<int> colors,
+                       Teuchos::RCP<femv_t> femv,
+                       bool recolor=false){
+      
+      //default values are taken from KokkosKernels_TestParameters.hpp
+      
+      int algorithm = pl->get<int>("Hybrid_algorithm",0);
+      int repeat = pl->get<int>("Hybrid_repeat",1); //probably not using this 
+                                                    //until we test performance
+      int chunk_size = pl->get<int>("Hybrid_chunk_size",-1); 
+      int shmemsize = pl->get<int>("Hybrid_shmemsize", 16128);
```

Comment:
Where do these magic constants come from?

Reply:
The defaults were taken from KokkosKernels_TestParameters.hpp, but it looks like the call succeeds without setting these parameters, so I will remove them.